File: Java_Joc_Windows_Android/AndroidStudioProject/core/src/base/menu/util/StateChangeDurations.java
package base.menu.util;

/**
 * Created by Adrin on 6/23/2016.
 */
public class StateChangeDurations {

    // String or String[]
    public Object startComposedStateID;
    // String or String[]
    public Object endComposedStateID;

    public float durationS;

    public StateChangeDurations(Object startComposedStateID, Object endComposedStateID, float durationS) {
        super();

        this.startComposedStateID = startComposedStateID;
        this.endComposedStateID = endComposedStateID;

        this.durationS = durationS;
    }

}

File: CPP_Joc_Windows_Android/SH_Client_Win_Cpp_Cmake/App/src/base/menu/util/StateChangeDurations.cpp
#include "StateChangeDurations.h"

StateChangeDurations::StateChangeDurations(std::string startComposedStateID, std::string endComposedStateID, float durationS)
    : durationS(durationS)
{
    this->startComposedStateID = UiId::newIdFromComposedId(startComposedStateID);
    this->endComposedStateID = UiId::newIdFromComposedId(endComposedStateID);
}

StateChangeDurations::~StateChangeDurations() {
    //void
}
